### PR TITLE
feat(module-tools): auto delete dist files by checking if outdir is root path

### PR DESCRIPTION
**Pull request by targeral**
### <samp>🤖 Generated by Copilot at c5d629a</samp>

Added a `noClear` option to the module-tools builder to avoid deleting the project when using a flat output structure. Updated the locale modules to include warning messages for this option in English and Chinese.

### <samp>🤖 Generated by Copilot at c5d629a</samp>

*  Add optional `options` parameter to `clearBuildConfigPaths` function to handle output directory being the same as project root directory ([link](https://github.com/web-infra-dev/modern.js/pull/4021/files?diff=unified&w=0#diff-20edc0d1347fe9a9ec86267371b7758bd0ae61f7420de1fec61116ed43c4da5dL11-R16), [link](https://github.com/web-infra-dev/modern.js/pull/4021/files?diff=unified&w=0#diff-20edc0d1347fe9a9ec86267371b7758bd0ae61f7420de1fec61116ed43c4da5dL20-R34))
*  Pass `noClear` flag and `projectAbsRootPath` value from `cmdOptions` and `context` to `clearBuildConfigPaths` function in `run` function of `builder` module ([link](https://github.com/web-infra-dev/modern.js/pull/4021/files?diff=unified&w=0#diff-fa01db21d10eb93a5a98d6545fbe49c63eab71da53447ace99b81543097aa95aL31-R34))
*  Add `clearRootPath` property to `EN_LOCALE` and `ZH_LOCALE` objects in `locale` module to store warning message for output directory being the same as project root directory ([link](https://github.com/web-infra-dev/modern.js/pull/4021/files?diff=unified&w=0#diff-044266080bf8162ec64961353e7f64de96a77d536bc9b2aacda435ba28b0c55cR65-R66), [link](https://github.com/web-infra-dev/modern.js/pull/4021/files?diff=unified&w=0#diff-6f0fa3593edef4f07b0b158eda75b877e7307c02189b5681f15beef8195f02f2R65-R66))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/solutions/module-tools/src/locale/en.ts [hunk 1]
```diff
@@ -62,5 +62,7 @@ export const EN_LOCALE = {
       abortOnError:
         'With the `dts.abortOnError` configuration currently turned off, type errors do not cause build failures, but they do not guarantee proper type file output',
     },
+    clearRootPath:
+      'It is detected that the outDir in the Edenx configuration is the same as the current project directory, and the current product directory will not be automatically deleted.',
```
Comment (chenjiahan) [inline packages/solutions/module-tools/src/locale/en.ts:66]: `Edenx` should not appear here, `Edenx configuration` -> `configuration`

File: packages/solutions/module-tools/src/builder/clear.ts [hunk 2]
```diff
@@ -8,15 +8,29 @@ export const clearDtsTemp = async () => {
...
+  const { noClear = false, projectAbsRootPath = process.cwd() } = options ?? {};
   const { fs } = await import('@modern-js/utils');
 
   if (noClear) {
     return;
   }
 
   for (const config of configs) {
-    await fs.remove(config.outDir);
+    if (projectAbsRootPath === config.outDir) {
```
Comment (chenjiahan) [inline packages/solutions/module-tools/src/builder/clear.ts:24]: Can `config.outDir` be a relative path?

File: packages/solutions/module-tools/src/locale/en.ts (hunk 1 above)
Comment (targeral) [reply packages/solutions/module-tools/src/locale/en.ts:66]: OK

File: packages/solutions/module-tools/src/builder/clear.ts (hunk 2 above)
Comment (targeral) [reply packages/solutions/module-tools/src/builder/clear.ts:24]: Module tools will convert all paths to absolute paths

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
close https://github.com/web-infra-dev/modern.js/issues/3995

**Pull request merged**